woodwork/statistics_utils.py

```diff
@@ -343,6 +343,136 @@ def _get_value_counts(dataframe, ascending=False, top_n=10, dropna=False):
+            - low_indices (list[int]): the corresponding index values for each of the lower outliers
+            - high_indices (list[int]): the corresponding index values for each of the upper outliers
+    """
+    if dd and isinstance(series, dd.Series):
+        series = series.compute()
+    if ks and isinstance(series, ks.Series):
+        series = series.to_pandas()
+
+    series = series.dropna()
+    if series.shape[0] == 0:
+        return
+
+    if quantiles is None:
+        quantiles = series.quantile([0.0, 0.25, 0.5, 0.75, 1.0]).to_dict()
+
+    if 0.25 in quantiles and 0.75 in quantiles:
+        low_bound, high_bound = _calculate_iqr_bounds(quantiles=quantiles)
+    else:
+        # if first and third quantile are not present, we have to calulate from the data
+        low_bound, high_bound = _calculate_iqr_bounds(series=series)
+
+    min = quantiles.get(0.0)
+    max = quantiles.get(1.0)
+    if min is not None and max is not None and low_bound <= min and high_bound >= max:
+        outliers_dict = {
+            "low_values": [],
+            "high_values": [],
+            "low_indices": [],
+            "high_indices": []
+        }
+    else:
+        # We've already removed nans and converted to pandas
+        outliers_dict = _get_outliers_for_column(series, low_bound, high_bound, convert_series=False)
+
+    return {'low_bound': low_bound,
+            'high_bound': high_bound,
+            'quantiles': quantiles,
+            **outliers_dict}
+
+
+def _get_outliers_for_column(series, low_bound=None, high_bound=None, convert_series=True):
+    """Gets the values of a series that lay above and below specified bounds.
+
+    Args:
+        series (Series): Data for which the outliers should be determined.
+        low_bound (float, optional): The number below which outliers lay. Is inclusive.
+        high_bound (float, optional): The number above which outliers lay. Is inclusive.
+
+    Note:
+        If neither or only one of low_bound or high_bound is passed in, the bounds will be calculated
+        using the IQR method.
+
+    Returns:
+        dict[str -> float,list[number]]: a dictionary containing outlier values and their corresponding indexes.
+            The following elements will be found in the dictionary:
+
+            - low_values (list[float, int]): the values of the lower outliers
+            - high_values (list[float, int]): the values of the upper outliers
+            - low_indices (list[int]): the corresponding index values for each of the lower outliers
+            - high_indices (list[int]): the corresponding index values for each of the upper outliers
+    """
+    if convert_series:
+        if dd and isinstance(series, dd.Series):
+            series = series.compute()
+        if ks and isinstance(series, ks.Series):
+            series = series.to_pandas()
+
+        series = series.dropna()
+        if series.shape[0] == 0:
+            return
```

Comment:
Should this be outside of the `if convert_series` block?

Reply:
it should be inside this block, though maybe `convert_series` isn't the best name. It's more about not repeating both the conversion to pandas as well as the removal of nans if we're calling the outlier function from within the box plot function, which will have already performed those operations. 

How about something like `transform_series`?

Reply:
Got it. I think `transform_series` might be a bit more indicative of what is happening.

Reply:
changing to `transfom_series`

Reply:
without separate `outlier_dict` and `box_plot_dict` methods, `transform_series` variable is no longer necessary, so it's been removed.